Description: Handle "nosym" keyword when detecting type of geometry orientation
Origin: http://openbabel.svn.sourceforge.net/viewvc/openbabel/openbabel/trunk/src/formats/gaussformat.cpp?r1=4836&r2=4835&pathrev=4836
Author: Michael Banck <mbanck@debian.org>
Last-Update: 2012-05-30
Bug: https://sourceforge.net/tracker/index.php?func=detail&aid=3495545&group_id=40728&atid=428740
Applied-Upstream: yes

--- a/src/formats/gaussformat.cpp	2011-09-15 20:44:53.000000000 +0200
+++ b/src/formats/gaussformat.cpp	2012-05-30 17:59:02.422708521 +0200
@@ -342,25 +342,30 @@
     }
 
     int i=0;
+    bool no_symmetry=false;
     char coords_type[25];
 
     //Prescan file to find second instance of "orientation:"
     //This will be the kind of coords used in the chk/fchk file
+    //Unless the "nosym" keyword has been requested
     while (ifs.getline(buffer,BUFF_SIZE))
       {
-        if (strstr(buffer,"orientation:") !=NULL)
-           i++;
-        if (i==2)
+        if (strstr(buffer,"Symmetry turned off by external request.") != NULL) 
+          {
+            // The "nosym" keyword has been requested
+            no_symmetry = true;
+          }
+        if (strstr(buffer,"orientation:") !=NULL) 
+          {
+            i++;
+            tokenize (vs, buffer);
+            strcpy (coords_type, vs[0].c_str());
+            strcat (coords_type, " orientation:");
+          }
+        if ((no_symmetry && i==1) || i==2)
            break;
-     }
-    if (!ifs.eof())
-    {
-    tokenize (vs, buffer);
-    strcpy (coords_type, vs[0].c_str());
-    strcat (coords_type, " orientation:");
+      }
     ifs.seekg(0);  //rewind
-    //    cout << coords_type << std::endl;
-    }
 
     mol.BeginModify();
     while (ifs.getline(buffer,BUFF_SIZE))
